src/ray/object_manager/object_manager.h

```diff
@@ -98,7 +100,7 @@ class ObjectManager : public ObjectManagerInterface {
   /// \param object_id The object's object id.
   /// \param client_id The remote node's client id.
```

Comment:
That would be interesting. In my observation in the object_manager_stress_test, this retry mechanism is triggered, so a long-term solution is necessary. 

By my understanding, we can first check the object_id in local_objects_ in the Push function. If it exists, the Push function continues. Otherwise, we can add this request to a map (as you described), and in NotifyDirectoryObjectAdd, the Push request will be re-triggered. It that correct?

Reply:
After talking to @raulchen and @eric-jj , I decide to change the code to waiting-list version.

Reply:
This is correct.